File: Python/7py 1/bng.py
"""
|--------------------------------------------------------------------------
| Band name generator
|--------------------------------------------------------------------------
|
"""

def band_name_generator(name):
   return f'The {name.capitalize()}' if name[0] != name[-1] else name.capitalize() + name[slice(1, len(name))]


def band_name_generator(name):
   return name.capitalize()+name[1:] if name[0]==name[-1] else 'The '+ name.capitalize()


def band_name_generator(name):
   return['The '+name.title(), name.title()+name[1:]][name[0]==name[-1]]


def band_name_generator(name):
   return ('The '+name, name+name[1:])[name.endswith(name[0])].title()


print(band_name_generator("knife"), "The Knife")
print(band_name_generator("tart"), "Tartart")
print(band_name_generator("sandles"), "Sandlesandles")
print(band_name_generator("bed"), "The Bed")
print(band_name_generator("qq"), "Qqq")

File: JavaScript/7js 1/bng.js
/*
|--------------------------------------------------------------------------
| Band name generator
|--------------------------------------------------------------------------
|
*/

function bandNameGenerator(str) {
   return str[0] == str.slice(-1) ?
      str[0].toUpperCase() + str.slice(1) + str.slice(1) :
      'The ' + str[0].toUpperCase() + str.slice(1)
}


function bandNameGenerator(s) {
   return /^(.).*\1$/.test(s) ?
      s[0].toUpperCase() + s.slice(1) + s.slice(1) :
      "The " + s[0].toUpperCase() + s.slice(1)
}


function bandNameGenerator(str) {
   return str.charAt(0) === str.charAt(str.length - 1) ?
      str.charAt(0).toUpperCase() + str.slice(1) + str.slice(1) :
      "The " + str.charAt(0).toUpperCase() + str.slice(1)
}


function bandNameGenerator(s) {
   return s[0] === s.slice(-1) ?
      s.replace(/^./, x => x.toUpperCase()) + s.slice(1) :
      `The ${s.replace(/^./, x => x.toUpperCase())}`
}


console.log(bandNameGenerator('knife'), 'The Knife');
console.log(bandNameGenerator('tart'), 'Tartart');
console.log(bandNameGenerator('sandles'), 'Sandlesandles');
console.log(bandNameGenerator('bed'), 'The Bed');